### DISCUSSION TITLE
Output to job summary?

### DISCUSSION THREAD

**Discussion by dtanner** [docker/build-push-action#1040] [Ideas]
Hi - have folks considered outputting a [step summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/)?
I'm transitioning to this action from a janky composite action I made, but a handy feature was that I output the destination image path and tags  to $GITHUB_STEP_SUMMARY. e.g. It would output something like this:
```shell
echo "### Docker Build" >> $GITHUB_STEP_SUMMARY
echo "Image pushed to <code>myregistry.company.com/my/image/path</code> with tags:" >> $GITHUB_STEP_SUMMARY
echo "- <code>sha-abc1234</code>" >> $GITHUB_STEP_SUMMARY
```

Any interest in adding something like this?

**Comment by sblack4** [comment]
You can use something like this:
```
      - name: Log Asset Info
        run: |
          echo "## Docker image metadata"
          echo '```' >> $GITHUB_STEP_SUMMARY
          echo '${{ steps.build-and-push.outputs.metadata }}' >> $GITHUB_STEP_SUMMARY
          echo '```' >> $GITHUB_STEP_SUMMARY
```

**Reply by crazy-max** [reply]
Latest release now generates a summary: https://github.com/docker/build-push-action/releases/tag/v6.0.0

**Reply by mfn** [reply]
That's nice but… does it really need to be on by default?

For all the … millions (?) uses of those github actions, isn't it excessive to generate all the artifacts suddenly?

Shouldn't it be rather opt-in?

**Reply by crazy-max** [reply]
cc @colinhemmings

**Reply by colinhemmings** [reply]
Hi @mfn, it's a great point and one we deliberated over. From our feedback, we felt there was sufficient value, and enabling the summary by default would allow more people to benefit from having this information available. It is also essential for us to keep these artifacts as small and efficient as possible for several reasons, one of which is not to waste a ton of space on redundant artifacts. 
We are constantly reviewing this early feedback and making iterations to make sure the feature is valuable and not disruptive. I'm happy to chat further on the topic. Do you feel like the summaries or artifacts are something you would opt out of?

**Reply by mfn** [reply]
> Do you feel like the summaries or artifacts are something you would opt out of?

Yeah, because not everyone building container needs this all the time.

Really, no idea how you weighted the arguments, but given the impact on basically anything docker does is "very big", this cries opt-in to me.
